internal/engine/experiment.go

```diff
@@ -188,10 +188,7 @@ func (e *Experiment) OpenReportContext(ctx context.Context) error {
 			Counter:      e.byteCounter,
 		},
 	}
-	if e.session.selectedProbeService == nil {
-		return errors.New("no probe services selected")
-	}
-	client, err := probeservices.NewClient(e.session, *e.session.selectedProbeService)
+	client, err := e.session.NewProbeServicesClient(ctx)
```

Comment:
We have a way of asking the session to give us a probe services client, so we should be using it.